public/app/plugins/panel/canvas/editor/PlacementEditor.tsx

```diff
@@ -48,6 +48,12 @@ export const PlacementEditor: FC<StandardEditorProps<any, CanvasEditorOptions, P
   const { options } = element;
   const { placement, constraint: layout } = options;
 
+  const reselectElementAfterChange = () => {
```

Comment:
After experimenting with a few different approaches, it seems most consistent / least problematic option (with a more significant refactor) is just to reselect the element with a slight delay after updating its layout option.